hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnectionChannelErrorHandler.java

```diff
@@ -45,7 +45,7 @@ public void onError(Channel channel, Throwable error) {
             TcpIpConnection connection = (TcpIpConnection) channel.attributeMap().get(TcpIpConnection.class);
             if (error instanceof EOFException) {
                 connection.close("Connection closed by the other side", error);
-            } else {
+            } else if (connection != null) {
```

Comment:
Do you know the scenario where the `connection` is null? are you sure it's impossible in the first branch?

Reply:
Good question! I saw the problem when a single node was started, but it had 2 members configured in the TCP join. It tried to connect to another member and failed with a `SocketException` as there was none.
I will change the code to cover both branches - just for sure.